src/core/contracts/subscription.test.ts

```diff
@@ -0,0 +1,174 @@
+import { keccak256 } from "js-sha3";
+
+import { batch, Announcement, dsnpBatchFilter } from "./announcement";
+import { batchAnnounceEvents } from "./subscription";
+import { setupConfig } from "../../test/sdkTestConfig";
+import { setupSnapshot } from "../../test/hardhatRPC";
+import { requireGetProvider } from "../../config";
+import { checkNumberOfFunctionCalls } from "../../test/utilities";
+import { BatchAnnounceCallbackArgs } from "./subscription";
+
+describe("subscription", () => {
+  setupSnapshot();
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.resetAllMocks();
```

Comment:
reset does everything clear does (+ extra)
```suggestion
    jest.resetAllMocks();
```

Actually, you don't reuse any mocks, so:

```suggestion
```

Reply:
oh yes. This I meant to clean up a while ago.